internal/core/data/v2/application/subscriber.go

```diff
@@ -85,3 +92,25 @@ func unmarshalPayload(envelope types.MessageEnvelope, target interface{}) error
 	}
 	return err
 }
+
+func validateEvent(messageTopic string, e dtos.Event) errors.EdgeX {
+	// Parse messageTopic by the pattern `edgex/events/<device-profile-name>/<device-name>/<source-name>`
+	fields := strings.Split(messageTopic, "/")
+	if len(fields) != 5 {
+		return errors.NewCommonEdgeX(errors.KindContractInvalid, fmt.Sprintf("invalid messag topic %s", messageTopic), nil)
+	}
+	profileName := fields[2]
+	deviceName := fields[3]
+	sourceName := fields[4]
+	// Check whether the event fields match the message topic
+	if e.ProfileName != profileName {
+		return errors.NewCommonEdgeX(errors.KindContractInvalid, fmt.Sprintf("event's profileName %s mismatches %s", e.ProfileName, profileName), nil)
+	}
+	if e.DeviceName != deviceName {
+		return errors.NewCommonEdgeX(errors.KindContractInvalid, fmt.Sprintf("event's deviceName %s mismatches %s", e.DeviceName, deviceName), nil)
+	}
+	if e.SourceName != sourceName {
+		return errors.NewCommonEdgeX(errors.KindContractInvalid, fmt.Sprintf("event's sourceName %s mismatches %s", e.SourceName, sourceName), nil)
```

Comment:
add context about name in topic like `"event's sourceName %s mismatches with name  %s received in topic" for each of these.

Reply:
Modified the error message.